nixos/modules/services/networking/netbird-server.nix

```diff
@@ -0,0 +1,343 @@
+{ config, lib, pkgs, ... }:
+let
+  inherit (lib)
+    literalExpression
+    maintainers
+    mkDefault
+    mkEnableOption
+    mkIf
+    mkOption
+    mkPackageOption
+    types
+    ;
+
+  cfg = config.services.netbird-server;
+
+  stateDir = "/var/lib/netbird-mgmt";
+
+  settingsFormat = pkgs.formats.keyValue { };
+  managementFormat = pkgs.formats.json { };
+
+  settingsFile = settingsFormat.generate "setup.env" (builtins.mapAttrs
+    (_: val:
+      if builtins.isList val then
+        ''"${builtins.concatStringsSep " " val}"''
+      else
+        val)
+    cfg.settings);
+  managementFile = managementFormat.generate "config.json" cfg.managementConfig;
+in
+{
+  meta.maintainers = with maintainers; [ thubrecht ];
```

Comment:
There should be a `meta.doc`, explaining how to actually use this module. There are a lot of options and it's not clear what a working "production"-like configuration should look like. A NixOS test is also desirable.

Reply:
I have added a bit of documentation and also reduced the amount of settings that the end user has to provide by default. I don't have the time right now to write a complicated test, as I believe we need to setup multiple nodes, one being an sso and try adding nodes to the network in the test and until the start of september I will be a bit busy.